plugin/pkg/scheduler/scheduler.go

```diff
@@ -120,11 +130,18 @@ func (s *Scheduler) scheduleOne() {
 		}
 
 		bindingStart := time.Now()
+		// If binding succeded then PodScheduled condition will be updated in apiserver so that
+		// it's atomic with setting host.
 		err := s.config.Binder.Bind(b)
 		if err != nil {
 			glog.V(1).Infof("Failed to bind pod: %+v", err)
 			s.config.Error(pod, err)
 			s.config.Recorder.Eventf(pod, api.EventTypeNormal, "FailedScheduling", "Binding rejected: %v", err)
+			s.config.PodConditionUpdater.Update(pod, &api.PodCondition{
+				Type:   api.PodScheduled,
+				Status: api.ConditionFalse,
+				Reason: "BindingRejected",
```

Comment:
Similar with above comment, an extra field with error message to tell why BindingRejected would be helpful.